pkg/service/did/storage.go

```diff
@@ -92,14 +134,27 @@ func (ds *Storage) GetDIDs(ctx context.Context, method string) ([]StoredDID, err
 		logrus.Infof("no DIDs found for method: %s", method)
 		return nil, nil
 	}
-	var stored []StoredDID
+
+	out := make([]StoredDID, 0, len(gotDIDs))
 	for _, didBytes := range gotDIDs {
-		var nextDID StoredDID
+		nextDID := reflect.New(reflect.TypeOf(outType).Elem()).Interface()
```

Comment:
this could be computationally really slow? (maybe not)

Reply:
yeah, depending on how large the set of DIDs is this could be bad. this will mostly be fine with #349